### PR TITLE
Update polynomial zonotope, add interface

### PR DESCRIPTION
**Pull request by mforets**

### CODE REVIEW

**Comment by mforets** [comment]
cc: @lucaferranti

File: src/Interfaces/AbstractPolynomialZonotope.jl [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+```jldoctest; setup = :(using LazySets: subtypes)
+julia> subtypes(AbstractPolynomialZonotope)
+1-element Vector{Any}:
+ DensePolynomialZonotope
+```
+"""
+abstract type AbstractPolynomialZonotope{N} end
+
+isconvextype(::Type{<:AbstractPolynomialZonotope}) = false
```
Comment (schillic) [inline src/Interfaces/AbstractPolynomialZonotope.jl:18]: To match the documentation:
```suggestion
isconvextype(::Type{<:AbstractPolynomialZonotope}) = false
isboundedtype(::Type{<:AbstractPolynomialZonotope}) = true
```

File: src/Sets/DensePolynomialZonotope.jl [hunk 2]
```diff
@@ -83,11 +86,25 @@ struct PolynomialZonotope{N, VT, VMT, MT}
...
 
 """
-    dim(pz::PolynomialZonotope)
+    PolynomialZonotope = DensePolynomialZonotope
+
+Alias for `DensePolynomialZonotope`.
+
+### Notes
+
+To review after LazySets#1543, which should use [1].
```
Comment (schillic) [inline src/Sets/DensePolynomialZonotope.jl:99]:
```suggestion
To be reviewed after LazySets#1543 (should use [1]).
```

**Review by schillic — APPROVED** [review]

File: src/Interfaces/AbstractPolynomialZonotope.jl [hunk 3]
```diff
@@ -0,0 +1,19 @@
+"""
```
Comment (schillic) [inline src/Interfaces/AbstractPolynomialZonotope.jl:1]: This file is currently not included.

**Pull request merged**